feat(billing): Add formatting configuration to DATA_CATEGORY_INFO formatting

dashed opened the pull request:
Closes https://linear.app/getsentry/issue/BIL-1812/stack-2-migrate-chart-functions-to-use-formatting-config
Continues from https://github.com/getsentry/sentry/pull/104023

- Update `mapReservedToChart` to use `formatting.reservedMultiplier` instead of `isByteCategory()` check with `GIGABYTE` constant
- Update `formatProjected` to use `formatting.projectedAbbreviated` instead of hardcoded `category !== DataCategory.ATTACHMENTS` check
- Export `mapReservedToChart` and add unit tests for multiplier logic

dashed committed:
feat(billing): Migrate chart functions to use DATA_CATEGORY_INFO formatting

- Update mapReservedToChart to use formatting.reservedMultiplier
  instead of isByteCategory() check with GIGABYTE constant
- Update formatProjected to use formatting.projectedAbbreviated
  instead of hardcoded DataCategory.ATTACHMENTS check
- Export mapReservedToChart and add unit tests for multiplier logic

Part of BIL-952: Move formatting functions to DATA_CATEGORY_INFO

isabellaenriquez commented on static/gsApp/views/subscriptionPage/reservedUsageChart.spec.tsx:
```diff
@@ -62,6 +65,48 @@ describe('mapStatsToChart', () => {
 });
 
+describe('mapReservedToChart', () => {
+  it('should apply GIGABYTE multiplier for byte categories (attachments)', () => {
```
nit: you could just cut down the tests to three for byte categories, duration categories, and count categories (plus the two edges cases for unlimited and null) since there isn't actually any differentiation between individual categories within those subsets

dashed committed:
refactor(billing): Consolidate redundant mapReservedToChart tests

Remove duplicate tests for categories that share the same multiplier logic.
Each multiplier type (byte, duration, count) now has a single representative test.
